### PR TITLE
Some improvement of CHE-1245

### PR DESCRIPTION
**Pull request by mmorhun**
Improve code which works with docker authentication

### CODE REVIEW

**Comment by mmorhun** [comment]
@garagatyi take a look please

File: plugins/plugin-docker/che-plugin-docker-client/src/main/java/org/eclipse/che/plugin/docker/client/DockerRegistryAuthResolver.java [hunk 1]
```diff
@@ -10,32 +10,36 @@
...
 
-    public static final ImmutableSet<String> DEFAULT_REGISTRY = ImmutableSet.of("",
-                                                                                "docker.io",
-                                                                                "index.docker.io",
-                                                                                "https://index.docker.io",
-                                                                                "https://index.docker.io/v1",
-                                                                                "https://index.docker.io/v1/");
-    public static final String DEFAULT_REGISTRY_VALUE = "docker.io";
+    public static final Set<String> DEFAULT_REGISTRY_SYNONYMS = Collections.unmodifiableSet(
+            com.google.common.collect.Sets.newHashSet(null,
```
Comment (garagatyi) [inline plugins/plugin-docker/che-plugin-docker-client/src/main/java/org/eclipse/che/plugin/docker/client/DockerRegistryAuthResolver.java]: Can you avoid usage of FQN here?

File: plugins/plugin-docker/che-plugin-docker-client/src/main/java/org/eclipse/che/plugin/docker/client/DockerRegistryAuthResolver.java [hunk 2]
```diff
@@ -45,101 +49,63 @@ public DockerRegistryAuthResolver(InitialAuthConfig initialAuthConfig) {
...
+            authConfigs.putAll(paramAuthConfigs.getConfigs());
         }
 
-        if (paramsAuthConfig != null) {
-            for(Map.Entry<String, AuthConfig> entry : paramsAuthConfig.entrySet()) {
-                AuthConfig value = entry.getValue();
-                authConfigs.put(entry.getKey(),
-                                new XRegistryAuthUnit(value.getUsername(), value.getPassword()));
-            }
+        if (authConfigs.get("docker.io") != null) {
```
Comment (garagatyi) [inline plugins/plugin-docker/che-plugin-docker-client/src/main/java/org/eclipse/che/plugin/docker/client/DockerRegistryAuthResolver.java]: please add comment that will describe what this code does

File: plugins/plugin-docker/che-plugin-docker-client/src/main/java/org/eclipse/che/plugin/docker/client/DockerRegistryAuthResolver.java [hunk 3]
```diff
@@ -45,101 +49,63 @@ public DockerRegistryAuthResolver(InitialAuthConfig initialAuthConfig) {
...
+            AuthConfig authConfig = authConfigs.get("docker.io");
```
Comment (garagatyi) [inline plugins/plugin-docker/che-plugin-docker-client/src/main/java/org/eclipse/che/plugin/docker/client/DockerRegistryAuthResolver.java]:
```
AuthConfig abnormalDefaultRegistryAuthConfig = authConfigs.remove("docker.io");
if (abnormalDefaultRegistryAuthConfig != null) {
    authConfigs.put(DEFAULT_REGISTRY, abnormalDefaultRegistryAuthConfig);
}
```

File: plugins/plugin-docker/che-plugin-docker-machine/src/main/java/org/eclipse/che/plugin/docker/machine/DockerInstance.java [hunk 4]
```diff
@@ -91,7 +89,7 @@
     private final String                                      image;
     private final LineConsumer                                outputConsumer;
     private final String                                      registry;
-    private final String                                      username;
+    private final String                                      namespace;
```
Comment (garagatyi) [inline plugins/plugin-docker/che-plugin-docker-machine/src/main/java/org/eclipse/che/plugin/docker/machine/DockerInstance.java]: Rename var to hint what for it is used

File: assembly/assembly-wsmaster-war/src/main/webapp/WEB-INF/classes/codenvy/che.properties [hunk 5]
```diff
@@ -69,6 +69,7 @@ schedule.core_pool_size=10
 ### that are constructed when the workspace is started. The images used to generate containers
 ### for the machines can come from DockerHub or a private Docker registry.
 machine.docker.registry=${CHE_REGISTRY_HOST}:5000
+machine.docker.registry.namespace=NULL
```
Comment (garagatyi) [inline assembly/assembly-wsmaster-war/src/main/webapp/WEB-INF/classes/codenvy/che.properties]: Rename property to show its purpose
machine.docker.snapshot.registry_namespace=NULL
I would also want to rename property 1 line above but it is not backward compatible

**Comment by garagatyi** [comment]
LGTM

**Comment by garagatyi** [comment]
@skabashnyuk Please take a look

**Comment by garagatyi** [comment]
@riuvshin In which version can we merge that PR?

**Comment by riuvshin** [comment]
I think it is OK to merge this to 4.4.0.

**Comment by skabashnyuk** [comment]
ok

**Pull request merged**